SwrveSDK/SDK/Talk/SwrveTriggerCondition.m

```diff
@@ -39,12 +39,19 @@ - (BOOL) hasFulfilledCondition:(NSDictionary *)payload {
     }
     
     NSArray *payloadKeys = [payload allKeys];
-    
     if([payloadKeys containsObject:_key]) {
-        
         if([payload objectForKey:_key] != [NSNull null]) {
-            
-            NSString *payloadValue = [payload objectForKey:_key];
+            id payloadObject = [payload objectForKey:_key];
+            NSString *payloadValue = nil;
+            if ([payloadObject isKindOfClass:NSString.class]) {
+                if ([payloadObject respondsToSelector:@selector(stringValue)]) {
```

Comment:
~~ if the value is a member of NSString class, why do we need an additional if statement here? we can parse it directly to NSString and should autobox all the other stuff AFAIK